zeek/Shaping-Zeek-NDJSON.md

```diff
@@ -0,0 +1,210 @@
+# Shaping Zeek NDJSON
+
+- [Summary](#summary)
+- [Reference Shaper Contents](#reference-shaper-contents)
+  * [Leading Type Definitions](#leading-type-definitions)
+  * [Type Definitions Per Zeek Log `_path`](#type-definitions-per-zeek-log-_path)
+  * [Mapping From `_path` Values to Types](#mapping-from-_path-values-to-types)
+  * [Zed Pipeline](#zed-pipeline)
+- [Invoking the Shaper From `zq`](#invoking-the-shaper-from-zq)
+- [Zeek Version/Configuration](#zeek-versionconfiguration)
+- [Importing Shaped Data Into Brim](#importing-shaped-data-into-brim)
+- [Contact us!](#contact-us)
+
+> **Note:** This document describes functionality that's available in Zed
+> `v0.30.0` and newer (hence also [Brim](https://github.com/brimdata/brim)
+> `v0.25.0` and newer). If you're looking for docs regarding the legacy
+>`types.json` approach that was used in Zed [`v0.29.0`](https://github.com/brimdata/zed/releases/tag/v0.29.0)
+> (or Brim [`0.24.0`](https://github.com/brimdata/brim/releases/tag/v0.24.0))
+> and older you can find it [here](https://github.com/brimdata/zed/blob/v0.29.0/zeek/README.md).
+
+# Summary
+
+As described in [Reading Zeek Log Formats](Reading-Zeek-Log-Formats.md),
+logs output by Zeek in NDJSON format lose much of their rich data typing that
+was originally present inside Zeek. This detail can be restored using a Zed
+shaper, such as the reference example [`shaper.zed`](shaper.zed)
+that can be found in this directory of the repository.
+
+A full description of all that's possible with shapers is beyond the scope of
+this doc. However, this example for shaping Zeek NDJSON is quite simple and
+is described below.
+
+# Reference Shaper Contents
+
+The example `shaper.zed` may seem large, but ultimately it follows a
+fairly simple pattern that repeats across the many [Zeek log types](https://docs.zeek.org/en/master/script-reference/log-files.html).
+
+## Leading Type Definitions
+
+The top three lines define types that are referenced further below in the main
+portion of the Zed.
```

Comment:
Zed is used as an adjective almost everywhere else, so I think this reads more smoothly if used the same way here.
```suggestion
portion of the Zed shaper.
```